src/services/narrowing.ts

```diff
@@ -131,19 +131,12 @@ export function narrowFilterOperator(op: string): LabelFilterOp | NumericFilterO
   }
 }
 
-export function narrowStringsArray(param: unknown): string[] | null {
-  try {
-    // Attempt to parse the string as JSON
-    const parsedParam = JSON.parse(decodeURIComponent(param as string));
-    // Verify the parsed result is an array of strings
-    if (Array.isArray(parsedParam) && parsedParam.every((item) => typeof item === 'string')) {
-      return parsedParam;
-    }
-    return null;
-  } catch (e) {
-    // If parsing fails, it's not a valid JSON string
-    return null;
+export function narrowStringsArray(array: unknown): string[] | null {
+  // Verify the parsed result is an array of strings
+  if (Array.isArray(array) && array.every((item) => typeof item === 'string')) {
+    return array;
   }
+  return null;
```

Comment:
Ah yes this is now like `unknownToStrings`. I will have to create some kind of function to parse url params which are just strings (not arrays) which is why I couldn't use `unknownToStrings`  or I can put the parsing logic back in the component logic `const parsedParam = JSON.parse(decodeURIComponent(param as string));`

Reply:
Ah, I will refactor if there's redundancy there.

> which is why I couldn't use unknownToStrings or I can put the parsing logic back in the component logic const parsedParam = JSON.parse(decodeURIComponent(param as string));

Yeah, I already made that change. The retrieval (searchParams.get(), localStorage.getItem()) or pasing (JSON.parse(), etc) is not a concern for narrowing. The consumer part (let's say a Scene) gets the `unknown` value from somewhere, and the narrowing part is in charge of determining that this unknown is of the expected type.